modal.js

```diff
@@ -0,0 +1,177 @@
+    this.logo_title = "";
+    this.logo_symbol = "";
+    this.nav = "";
+    this.linkHome = "";
+    this.linkServices = "";
+    this.linkPortfolio = "";
+    this.linkAbout = "";
+    this.linkContact = "";
+
+    this.createDomNode();
+    this.buildModal();
+    this.createModalContent();
+    this.gatherTogether();
+    //this.modalOpen();
+    //  this.modalClose();
+  }
+
+  createDomNode(node, element, ...classes) {
+    node = document.createElement(element);
+    node.classList.add(...classes);
+    return node;
+  }
+
+  buildModal() {
+    //overlay
+    this.overlay = this.createDomNode(this.overlay, "div", "overlay");
+
+    //modal
+    this.modal = this.createDomNode(this.modal, "div", "modal");
+  }
+
+  createModalContent() {
+    this.modalContent = this.createDomNode(
+      this.modalContent,
+      "div",
+      "modal__content"
+    );
+    //Logo
+    this.logo = this.createDomNode(this.logo, "div", "modal__logo");
+    this.logo_title = this.createDomNode(
+      this.logo_title,
+      "h1",
+      "logo-title",
+      "modal__logo__title"
+    );
+    this.logo_title.innerText = "singolo";
+
+    this.logo_symbol = this.createDomNode(
+      this.logo_symbol,
+      "span",
+      "logo-symbol"
+    );
+    this.logo_symbol.innerText = "*";
+
+    this.logo.append(this.logo_title, this.logo_symbol);
+
+    //Navigation
+    this.nav = this.createDomNode(this.nav, "div", "modal__nav");
+
+    this.linkHome = this.createDomNode(this.linkHome, "a", "modal__nav__links");
+    this.linkHome.innerText = "home";
+    this.linkHome.setAttribute("href", "#home");
+
+    this.linkServices = this.createDomNode(
+      this.linkServices,
+      "a",
+      "modal__nav__links"
+    );
+    this.linkServices.innerText = "services";
+    this.linkServices.setAttribute("href", "#services");
+
+    this.linkPortfolio = this.createDomNode(
+      this.linkPortfolio,
+      "a",
+      "modal__nav__links"
+    );
+    this.linkPortfolio.innerText = "portfolio";
+    this.linkPortfolio.setAttribute("href", "#portfolio");
+
+    this.linkAbout = this.createDomNode(
+      this.linkAbout,
+      "a",
+      "modal__nav__links"
+    );
+    this.linkAbout.innerText = "about";
+    this.linkAbout.setAttribute("href", "#about");
+
+    this.linkContact = this.createDomNode(
+      this.linkContact,
+      "a",
+      "modal__nav__links"
+    );
+    this.linkContact.innerText = "contact";
+    this.linkContact.setAttribute("href", "#contact");
+
+    this.nav.append(
+      this.linkHome,
+      this.linkServices,
+      this.linkPortfolio,
+      this.linkAbout,
+      this.linkContact
+    );
+
+    //appending all to modalContent
+    this.modalContent.append(this.logo, this.nav);
+    return this.modalContent;
+  }
+
+  gatherTogether() {
+    this.modal.classList.add("modal-disappear");
+
+    this.modal.append(this.modalContent);
+    this.overlay.append(this.modal);
+    return this.overlay;
+  }
+
+  modalOpen() {
```

Comment:
the good code style is to name functions starting from the verb

Reply:
done!
changed to openModal and closeModal